mne/decoding/search_light.py

```diff
@@ -338,7 +358,10 @@ def _sl_score(estimators, X, y):
     """
     n_iter = X.shape[-1]
     for ii, est in enumerate(estimators):
-        _score = est.score(X[..., ii], y)
+        if scoring is not None:
+            _score = scoring(est, X[..., ii], y)
```

Comment:
your y is 2d and roc_auc requires y to be 1d or 1 column. You need to look over the columns of y.